src/backend/distributed/transaction/backend_data.c

```diff
@@ -998,6 +999,19 @@ citus_backend_gpid(PG_FUNCTION_ARGS)
 }
 
 
+/*
+ * citus_backend_nodeid similar to pg_backend_pid, but returns Citus
+ * assigned nodeid.
+ */
+Datum
```

Comment:
I guess we cannot use `SELECT citus_nodeid_for_gpid(citus_backend_gpid());` here because gpid might not be assigned?

Reply:
I didn't like that citus_backend_nodeid would return 99999999 if the node wasn't added to the metadata. So instead I changed citus_calculate_gpid to allow taking NULL for the first argument, which means that we're calculating for the local node.